actionsystemtest/lib/action_system_test.rb

```diff
@@ -0,0 +1,153 @@
+  # can change the driver, add Capybara settings, and other configuration for
+  # your system tests.
+  #
+  #   require "test_helper"
+  #
+  #   class ActionSystemTestCase < ActionSystemTest::Base
+  #     teardown do
+  #       take_failed_screenshot
+  #       Capybara.reset_sessions!
+  #     end
+  #   end
+  #
+  # By default, <tt>ActionSystemTest</tt> is driven by the Selenium driver, with
+  # the Chrome browser, on port 21800, and a browser size of 1400x1400.
+  #
+  # Changing the driver configuration options are easy. Let's say you want to use
+  # port 3000, and the Firefox browserinstead. In your +system_test_helper.rb+
+  # file add the following:
+  #
+  #   require "test_helper"
+  #
+  #   class ActionSystemTestCase < ActionSystemTest::Base
+  #     driven_by :selenium, using: :firefox, on: 3000
+  #
+  #     teardown do
+  #       take_failed_screenshot
+  #       Capybara.reset_sessions!
+  #     end
+  #   end
+  #
+  # +driven_by+ has a requried argument for the driver name. The keyword
+  # arguments are +:using+ for the browser (not applicable for headless drivers),
+  # +:on+ for port (the server is always Puma), and +:screen_size+ to change
+  # the size of the screen when taking screenshots.
+  #
+  # To use a headless driver, like Poltergeist, update your Gemfile to use
+  # Poltergeist instead of Selenium and then declare the driver name in the
+  # +system_test_helper.rb+ file. In this case you would leave out the +:using+
+  # option because the driver is headless.
+  #
+  #   require "test_helper"
+  #   require "capybara/poltergeist"
+  #
+  #   class ActionSystemTestCase < ActionSystemTest::Base
+  #     driven_by :poltergeist
+  #
+  #     teardown do
+  #       take_failed_screenshot
+  #       Capybara.reset_sessions!
+  #     end
+  #   end
+  #
+  # Because <tt>ActionSystemTest</tt> is a shim between Capybara and Rails, any
+  # driver that is supported by Capybara is supported by Action System Test as
+  # long as you include the required gems and files.
+  include Capybara::DSL
+  include ActionSystemTest::TestHelpers::ScreenshotHelper
+
+  class Base < ActionDispatch::IntegrationTest
+    include ActionSystemTest
+
+    def self.start_application # :nodoc:
+      Capybara.app = Rack::Builder.new do
+        map "/" do
+          run Rails.application
+        end
+      end
+    end
+
+    # Action System Test configuration options
+    #
+    # The defaults settings are Selenium, using Chrome, on port 21800, with a
+    # screen size of 1400x1400.
+    #
+    # Examples:
+    #
+    #   driven_by :poltergeist
+    #
+    #   driven_by :selenium, using: :firefox, on: 3000
+    def self.driven_by(driver, using: :chrome, on: 21800, screen_size: [1400, 1400])
+      Driver.new(driver).run
+      Server.new(on).run
+      Browser.new(using, screen_size).run if selenium?(driver)
+    end
+
+    def self.selenium?(driver) # :nodoc:
```

Comment:
Maybe this is nit-picking... but wonder if name `selenium?` here is the "right" name, since the work of this method is to figure out whether the driver is headless or not  :)